packages/playwright-test/src/docker/docker.ts

```diff
@@ -172,15 +182,22 @@ export async function ensureContainerOrDie(): Promise<ContainerInfo> {
   return info;
 }
 
-export async function stopContainer() {
-  const containerId = await findRunningDockerContainerId();
-  if (!containerId)
+export async function stopAllContainers() {
+  const allContainers: Container[] | undefined = await getJSON('/containers/json');
+  if (!allContainers?.length)
     return;
-  await Promise.all([
+  const allImages: DockerImage[] | undefined = await getJSON('/images/json');
+  // Get all docker images from current and past Playwright versions.
+  const dockerImageIds = new Set(allImages?.filter(image => image.RepoTags?.some(repoTag => VRT_IMAGE_NAME_REGEX.test(repoTag))).map(image => image.Id) ?? []);
```

Comment:
Would be nice to only stop containers launched by Playwright itself. So add a meta label on run and filter by it.

Reply:
Agreed. Closed this in favor of https://github.com/microsoft/playwright/pull/17316